### PR TITLE
Front-end refactoring

### PR DESCRIPTION
**Pull request by reddigari**
- Move all API calls to functions in separate file. Allows other components to use `API.createNode()`, e.g., instead of making XHR requests and handling responses in the component. (Reduces footprint of Workspace component by almost half.)
    - All functions are async and return a promise that resolves to the expected JSON data, or rejects to the server's JSON error message
- Refactors the node menu components into their own file
- Minor diagram style tweaks

### CODE REVIEW

File: front-end/src/components/Workspace.js [hunk 1]
```diff
@@ -21,46 +22,14 @@ class Workspace extends React.Component {
...
-            const downloadData = await resp.json();
-            const dataStr = "data:text/json;charset=utf-8,"
-                + encodeURIComponent(JSON.stringify(downloadData));
-            const anchor = document.createElement("a")
-            anchor.href = dataStr;
-            anchor.download = downloadData.filename || "diagram.json";
-            anchor.click();
-            anchor.remove();
-        }
+        API.getNodes().then(nodes => this.setState({nodes: nodes}));
```
Comment (reelmatt) [inline front-end/src/components/Workspace.js:31]: Missing `.catch()` on these two calls

File: front-end/src/components/Workspace.js [hunk 2]
```diff
@@ -74,67 +43,43 @@ class Workspace extends React.Component {
...
 
     /**
-     * Remove all nodes from diagram
+     * Remove all nodes from diagram and initialize new workflow on server
      */
     clear() {
-        if (window.confirm("Clear diagram?")) {
+        if (window.confirm("Clear diagram? You will lose all work.")) {
             this.model.getNodes().forEach(n => n.remove());
+            API.initWorkflow().then(graph => console.log(graph));
```
Comment (reelmatt) [inline front-end/src/components/Workspace.js:51]: And a `.catch()` here

**Comment by reddigari** [comment]
@reelmatt I think you're right --- I wrestled with that idea, and the nested and chained `then`/`catches` always look kinda gross, but the functions repeat themselves so much that it's probably worthwhile. The complicated ones can just handle the data after the call to `ourFetch`.

Updates coming shortly.

**Comment by reddigari** [comment]
I think we're good, but if someone wants to double-check that I didn't break any functionality with the last commits before it's merged, that'd be great.

**Comment by reelmatt** [comment]
Looks good to me!

**Pull request merged**